File: brain_api/schemas/dtos/recommendation.py
from pydantic import BaseModel, Field


class RecommendationRequest(BaseModel):
    """Request schema for charging recommendations."""

    latitude: float = Field(..., ge=-90, le=90, description="Vehicle latitude")
    longitude: float = Field(..., ge=-180, le=180, description="Vehicle longitude")
    battery_level: int = Field(
        ..., ge=0, le=100, description="Current battery level (%)"
    )


class RecommendationResponse(BaseModel):
    """Response schema for charging recommendations."""

    hub_id: str = Field(..., description="Recommended hub ID")
    node_id: str = Field(..., description="Recommended node ID")
    hub_latitude: float = Field(..., description="Hub latitude")
    hub_longitude: float = Field(..., description="Hub longitude")
    distance_km: float = Field(..., description="Distance from vehicle to hub in km")
    estimated_wait_time_min: int = Field(
        ..., description="Estimated wait time in minutes"
    )
    available_power_kw: float = Field(..., description="Available charging power in kW")

File: mobile_app/lib/models/recommendation.dart
class Recommendation {
  final String hubId;
  final String nodeId;
  final double hubLatitude;
  final double hubLongitude;
  final double distanceKm;
  final int estimatedWaitTimeMin;
  final double availablePowerKw;

  Recommendation({
    required this.hubId,
    required this.nodeId,
    required this.hubLatitude,
    required this.hubLongitude,
    required this.distanceKm,
    required this.estimatedWaitTimeMin,
    required this.availablePowerKw,
  });

  factory Recommendation.fromJson(Map<String, dynamic> json) {
    return Recommendation(
      hubId: json['hub_id'],
      nodeId: json['node_id'],
      hubLatitude: json['hub_latitude'].toDouble(),
      hubLongitude: json['hub_longitude'].toDouble(),
      distanceKm: json['distance_km'].toDouble(),
      estimatedWaitTimeMin: json['estimated_wait_time_min'],
      availablePowerKw: json['available_power_kw'].toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'hub_id': hubId,
      'node_id': nodeId,
      'hub_latitude': hubLatitude,
      'hub_longitude': hubLongitude,
      'distance_km': distanceKm,
      'estimated_wait_time_min': estimatedWaitTimeMin,
      'available_power_kw': availablePowerKw,
    };
  }
}
